cmd/oras/internal/display/metadata/text/attach.go

```diff
@@ -42,21 +41,21 @@ func NewAttachHandler(printer *output.Printer) metadata.AttachHandler {
 
 // OnAttached implements AttachHandler.
 func (ah *AttachHandler) OnAttached(target *option.Target, root ocispec.Descriptor, subject ocispec.Descriptor) {
-	ah.target = target
+	ah.subjectRefByDigest = target.AnnotatedReference()
+	if !strings.HasSuffix(target.RawReference, subject.Digest.String()) {
+		// use subject digest instead of tag
+		newTarget := *target
+		newTarget.RawReference = fmt.Sprintf("%s@%s", target.Path, subject.Digest)
+		ah.subjectRefByDigest = newTarget.AnnotatedReference()
+	}
```

Comment:
nit:
```suggestion
	if strings.HasSuffix(target.RawReference, subject.Digest.String()) {
		ah.subjectRefByDigest = target.AnnotatedReference()
	} else {
		// use subject digest instead of tag
		newTarget := *target
		newTarget.RawReference = fmt.Sprintf("%s@%s", target.Path, subject.Digest)
		ah.subjectRefByDigest = newTarget.AnnotatedReference()
	}
```